<a href="https://colab.research.google.com/github/santebiondifc/Corso/blob/main/Es01_.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# escolha o arquivo:



In [2]:
from google.colab import files
uploaded = files.upload ()

Saving matriculas_escola.csv to matriculas_escola.csv


Importare Pandas e matplotlib

In [3]:
import pandas as pd
import matplotlib.pyplot as plt

escola = pd.read_csv("matriculas_escola.csv")
print ("Linee e colonne",escola.shape)
escola.head()

Linee e colonne (384, 8)


,matricula_id,aluno,responsavel,curso,turno,situacao,mensalidade,data_matricula
0,5001,Nathalia Kunz,Gabriela Fagundes,Técnico em Segurança do Trabalho,Manhã,Concluído,310.0,2026-02-03
1,5002,Sabrina Brum,Marcos Carvalho,Técnico em Contabilidade,Tarde,CONCLUIDO,310.0,2025-02-20
2,5003,Tiago Schneider,Nathalia Machado Rocha,Técnico em Administração,Noite,EVADIDO,340.0,2026-04-23
3,5004,Elisa Lima,Nelson Fontoura,Técnico em Administração,Noite,CURSANDO,330.0,2025-02-13
4,5005,Alexandre Brum,Leonardo Brum,Enfermagem,Noite,CONCLUIDO,490.0,2025-03-18


Pulizia del file con .isna().mean()*100).round(1)

In [4]:
mancanti = (escola.isna().mean()*100).round(1)
print(mancanti.sort_values(ascending=False))

responsavel       4.7
matricula_id      0.0
aluno             0.0
curso             0.0
turno             0.0
situacao          0.0
mensalidade       0.0
data_matricula    0.0
dtype: float64


Pulizia "curso"

In [5]:
print(escola["curso"].value_counts())

curso
Técnico em Administração             72
Técnico em Informática               44
Técnico em Segurança do Trabalho     38
TECNICO EM ADMINISTRACAO             32
Técnico em Enfermagem                30
Técnico em Estética                  24
Técnico em Logística                 23
Técnico em Contabilidade             20
SEGURANCA DO TRABALHO                14
TECNICO EM INFORMATICA               14
Enfermagem                           14
TECNICO EM CONTABILIDADE             11
TECNICO EM LOGISTICA                 10
TECNICO EM ESTETICA                   8
TECNICO EM ENFERMAGEM                 6
TECNICO EM ADMINISTRACAO              5
Técnico em Administração              3
Técnico em Informática                3
Técnico em Logística                  2
TECNICO EM LOGISTICA                  2
Técnico em Enfermagem                 2
Enfermagem                            1
TECNICO EM ENFERMAGEM                 1
Técnico em Segurança do Trabalho      1
Técnico em Contabilidade          

puliamo tutte le colonne

In [6]:
escola["aluno"] = escola["aluno"].str.strip().str.upper()
escola["responsavel"] = escola["responsavel"].str.strip().str.upper()
escola["curso"] = escola["curso"].str.strip().str.upper()
escola["turno"] = escola["turno"].str.strip().str.upper()
escola["situacao"] = escola["situacao"].str.strip().str.upper()
escola["situacao"] = escola["situacao"].str.strip().str.upper()
print(escola["curso"].value_counts())

curso
TÉCNICO EM ADMINISTRAÇÃO            75
TÉCNICO EM INFORMÁTICA              47
TÉCNICO EM SEGURANÇA DO TRABALHO    39
TECNICO EM ADMINISTRACAO            37
TÉCNICO EM ENFERMAGEM               32
TÉCNICO EM ESTÉTICA                 25
TÉCNICO EM LOGÍSTICA                25
TÉCNICO EM CONTABILIDADE            21
ENFERMAGEM                          15
TECNICO EM INFORMATICA              15
SEGURANCA DO TRABALHO               14
TECNICO EM LOGISTICA                12
TECNICO EM CONTABILIDADE            11
TECNICO EM ESTETICA                  9
TECNICO EM ENFERMAGEM                7
Name: count, dtype: int64


Togliere i doppioni

In [7]:
de_para_curso={
    "TÉCNICO EM ADMINISTRAÇÃO":"Tecnico em aministracao",
    "TÉCNICO EM INFORMÁTICA":"Tecnico em informatica",
    "TÉCNICO EM SEGURANÇA DO TRABALHO":"Tecnico em seguranca do trabalho",
    "TECNICO EM ADMINISTRACAO":"Tecnico em aministracao",
    "TÉCNICO EM ENFERMAGEM":"Tecnico em enfermagem",
    "TÉCNICO EM ESTÉTICA":"Tecnico em estetica",
    "TÉCNICO EM LOGÍSTICA":"Tecnico em logistica",
    "TÉCNICO EM CONTABILIDADE":"Tecnico em contabilidade",
    "ENFERMAGEM":"Tecnico em enfermagem",
    "TECNICO EM INFORMATICA":"Tecnico em informatica",
    "SEGURANCA DO TRABALHO":"Tecnico em seguranca do trabalho",
    "TECNICO EM LOGISTICA":"Tecnico em logistica",
    "TECNICO EM CONTABILIDADE":"Tecnico em contabilidade",
    "TECNICO EM ESTETICA":"Tecnico em estetica",
    "TECNICO EM ENFERMAGEM":"Tecnico em enfermagem"
}
escola["curso"] = escola["curso"].replace(de_para_curso)
print(escola["curso"].value_counts())

curso
Tecnico em aministracao             112
Tecnico em informatica               62
Tecnico em enfermagem                54
Tecnico em seguranca do trabalho     53
Tecnico em logistica                 37
Tecnico em estetica                  34
Tecnico em contabilidade             32
Name: count, dtype: int64


In [8]:
print(escola["turno"].value_counts())

turno
NOITE    188
MANHÃ    123
TARDE     73
Name: count, dtype: int64


turno

In [9]:
de_para_turno = {
    "NOITE":"Noite",
    "MANHÃ":"Manha",
    "TARDE":"Tarde"
}
escola["turno"] = escola["turno"].replace(de_para_turno)
print(escola["turno"].value_counts())

turno
Noite    188
Manha    123
Tarde     73
Name: count, dtype: int64


situacao


In [10]:
print(escola["situacao"].value_counts())

situacao
CURSANDO     168
EVADIDO       97
CONCLUÍDO     77
CONCLUIDO     42
Name: count, dtype: int64


In [11]:
de_para_situacao = {
    "CURSANDO":"Cursando",
    "EVADIDO":"Evadido",
    "CONCLUÍDO":"Concluido",
    "CONCLUIDO":"Concluido"
}
escola["situacao"] = escola["situacao"].replace(de_para_situacao)
print(escola["situacao"].value_counts())

situacao
Cursando     168
Concluido    119
Evadido       97
Name: count, dtype: int64


duplicatas — a mesma pessoa, no mesmo curso, na mesma data: é rematrícula digitada duas vezes.

In [12]:
escola["data_matricula"]=pd.to_datetime(escola["data_matricula"], errors="coerce")
print(escola["data_matricula"].head())

0   2026-02-03
1   2025-02-20
2   2026-04-23
3   2025-02-13
4   2025-03-18
Name: data_matricula, dtype: datetime64[ns]


In [13]:
duplicada=escola.duplicated(subset=["aluno", "data_matricula","curso"])
nome_completo=escola["aluno"].str.split().str.len()>=2
tem_situacao=escola["situacao"].notna()
print("Base bruta:                ",len(escola))
print("Sem duplicatas:            ",(~duplicada).sum())
print("Com nome completo:         ",(~duplicada & nome_completo).sum())
print("Com situação:              ",(~duplicada & nome_completo & tem_situacao).sum())


Base bruta:                 384
Sem duplicatas:             360
Com nome completo:          360
Com situação:               360


Salvare il nuovo elaborato

In [14]:
trattato=escola[~duplicada & nome_completo & tem_situacao]
trattato.to_csv("escola_trattato.csv",index=False)
print("Base trattata:", len(trattato),"linee")


Base trattata: 360 linee


Le tre domande:
Qual è il tasso di abbandono scolastico? E come varia a seconda del corso di studi?

In [15]:
taxa_evasao = trattato["situacao"].eq("Evadido").mean() * 100

print("Total de alunos:", len(trattato))
print("Alunos evadidos:", trattato["situacao"].eq("Evadido").sum())
print("Taxa de abandono:", round(taxa_evasao, 1), "%")

Total de alunos: 360
Alunos evadidos: 88
Taxa de abandono: 24.4 %


In [16]:
taxa_por_curso = (
    trattato.groupby("curso")["situacao"]
    .apply(lambda x: x.eq("Evadido").mean() * 100)
    .round(1)
)

print(taxa_por_curso)

curso
Tecnico em aministracao             28.0
Tecnico em contabilidade            35.5
Tecnico em enfermagem               38.0
Tecnico em estetica                 18.8
Tecnico em informatica              17.5
Tecnico em logistica                20.0
Tecnico em seguranca do trabalho    10.4
Name: situacao, dtype: float64


Qual turno concentra a evasão? A escola tem mais aula à noite — isso ajuda ou atrapalha?

In [17]:
evasao_turno = (
    trattato.groupby("turno")["situacao"]
    .apply(lambda x: x.eq("Evadido").mean() * 100)
    .round(1)
)

print(evasao_turno)

turno
Manha    21.6
Noite    25.6
Tarde    26.5
Name: situacao, dtype: float64


Quantos alunos distintos a escola atendeu de fato no período?



In [18]:
alunos_distintos = trattato[
    trattato["situacao"].isin(["Cursando", "Concluido"])
]["aluno"].nunique()

print("Alunos distintos atendidos:", alunos_distintos)

Alunos distintos atendidos: 261


Responsavel

In [20]:
evadidos = trattato[trattato["situacao"] == "Evadido"]

print("Total evadidos:", len(evadidos))
print("Com responsável:", evadidos["responsavel"].notna().sum())
print("Sem responsável:", evadidos["responsavel"].isna().sum())
taxa_com_responsavel = evadidos["responsavel"].notna().mean() * 100

print(
    "Evadidos com responsável:",
    round(taxa_com_responsavel, 1),
    "%"
)

Total evadidos: 88
Com responsável: 82
Sem responsável: 6
Evadidos com responsável: 93.2 %


Associato ad ogni responsabile

In [22]:
evadidos_por_responsavel = (
    trattato[trattato["situacao"] == "Evadido"]
    .dropna(subset=["responsavel"])
    .groupby("responsavel")["aluno"]
    .nunique()
    .sort_values(ascending=False)
)

print(evadidos_por_responsavel)

responsavel
ADRIANA FAGUNDES             1
ADRIANA SOUZA                1
ALEXANDRE BARCELLOS          1
ALEXANDRE COSTA TRINDADE     1
ALEXANDRE FERREIRA LIMA      1
                            ..
VANESSA MARTINS              1
VINICIUS BORTOLOTTO          1
VINICIUS CARDOSO FONTOURA    1
VINICIUS KUNZ                1
VINICIUS TRINDADE ZANELLA    1
Name: aluno, Length: 82, dtype: int64


Il database originariamente contiene 384 elementi, dopo la pulizia eseguita sono rimasti 360.

Base bruta:                 384
Sem duplicatas:             360
Com nome completo:          360
Com situação:               360

La pipeline:
Importato il file csv
importato Pandas e Matplotlib
Verificato il contenuto del file
Verificato i mancanti: solo responsabile aveva un 4,7% di mancanti. Qui ho ritenuto ininfluente di conseguenza l’ho ignorato.
Ho pulito le colonne curso, turno, situacao, data e aluno.
A questo punto ho verificato quanti alunni rimanevano:

Base bruta:                 384
Sem duplicatas:             360
Com nome completo:          360
Com situação:               360

A questo punto avevo il tasso di alunni evasi

Total de alunos: 360
Alunos evadidos: 88
Taxa de abandono: 24.4 %

E per curso:
curso
Tecnico em aministracao             28.0
Tecnico em contabilidade            35.5
Tecnico em enfermagem               38.0
Tecnico em estetica                 18.8
Tecnico em informatica              17.5
Tecnico em logistica                20.0
Tecnico em seguranca do trabalho    10.4

Le discipline com piú evasi sono contabilità (35,5%), enfermagem (38%) e amministrazione (28%) mentre quella con meno evasi è sicurezza (10,4%).

La domanda successiva era per turno

Manha    21.6
Noite    25.6
Tarde    26.5

E qui invece non si rileva variazioni dal punto di vista del turno.

E infine la domanda finale è quanti aluni hanno frequentato la scuola.

Alunos distintos atendidos: 261

In base a quanto risultato mi concentrerei sulle tre discipline (contabilità, enfermagem e amministrazione) che hanno il maggior tasso di evasioni.


O banco de dados continha originalmente 384 itens; após a limpeza, restaram 360.

Base bruta: 384
Sem duplicatas: 360
Com nome completo: 360
Com localização: 360

O pipeline:
Importação do arquivo CSV
Importação do Pandas e Matplotlib
Verificação do conteúdo dos arquivos
Verificação de dados faltantes: apenas o arquivo em questão apresentava 4,7% de dados faltantes. Considerei esse valor irrelevante e o ignorei.

Limpeza das colunas Curso, turno, situação, data e aluno.

Neste ponto, verifiquei quantos alunos restavam:

Base bruta: 384
Sem duplicados: 360
Com nome completo: 360
Situação: 360

Neste ponto, eu tinha a taxa de alunos evadidos:

Número total: 360
Sozinhos evadidos: 88
Taxa de abandono: 24,4%

E, aliás:
curso
Técnico administrativo 28,0%
Técnico contábil 35,5%
Técnico em enfermagem 38,0%
Técnico e estético 18,8%
Técnico em TI 17,5%
Técnico em logística 20,0%
Técnico em segurança do trabalho 10,4%

As disciplinas com maior evasão são contabilidade (35,5%), gestão (38%) e administração (28%), enquanto a com menor evasão é segurança (10,4%).

A próxima pergunta foi por turno.

Manhã 21,6
Noite 25,6
Tarde 26,5

E aqui, no entanto, não há alterações nos turnos.

E, finalmente, a última pergunta é quantos alunos frequentaram a escola.

Número total de alunos presentes: 261

Verifiquei com os gerentes se poderia haver alguma relação.

O resultado é irrelevante, já que todos têm 1 aluno faltoso.

Responsável
ADRIANA FAGUNDES 1
ADRIANA SOUZA1
ALEXANDRE BARCELLOS 1
ALEXANDRE COSTA TRINDADE 1
ALEXANDRE FERREIRA LIMA 1

VANESSA MARTINS 1
VINICIUS BORTOLOTTO 1
VINICIUS CARDOSO FONTOURA 1
VINICIUS KUNZ 1
VINICIUS TRINDADE ZANELLA 1

Leitura gerencial

Com base nos resultados, eu me concentraria nas três disciplinas (contabilidade, enfermagem e administração) que apresentam as maiores taxas de evasão.